src/mantine-core/src/Floating/FloatingArrow/get-arrow-position-styles.ts

```diff
@@ -111,20 +66,20 @@ export function getArrowPositionStyles({
   if (side === 'top') {
     return {
       ...baseStyles,
-      ...verticalSide(placement, arrowX, arrowOffset, dir),
+      left: arrowX,
       bottom: arrowPosition,
       borderTop: 0,
-      [dir === 'ltr' ? 'borderLeft' : 'borderRight']: 0,
+      borderLeft: 0,
     };
   }
 
   if (side === 'bottom') {
     return {
       ...baseStyles,
-      ...verticalSide(placement, arrowX, arrowOffset, dir),
+      left: arrowX,
       top: arrowPosition,
       borderBottom: 0,
-      [dir === 'ltr' ? 'borderRight' : 'borderLeft']: 0,
```

Comment:
This check was actually messing things up for RTL, simply removing it fixes things